connector.go

```
@@ -126,32 +129,38 @@ func (c *connector) listener(ctx *replication.ListenerContext) {
 		chunks := slices.ChunkWithSize[elasticsearch.Action](actions, batchSizeLimit)
 		lastChunkIndex := len(chunks) - 1
 		for idx, chunk := range chunks {
-			c.bulk.AddActions(ctx, msg.EventTime, chunk, msg.TableNamespace, msg.TableName, idx == lastChunkIndex)
+			c.bulk.AddActions(ctx, msg.EventTime, chunk, indexName, idx == lastChunkIndex)
 		}
 	} else {
-		c.bulk.AddActions(ctx, msg.EventTime, actions, msg.TableNamespace, msg.TableName, true)
+		c.bulk.AddActions(ctx, msg.EventTime, actions, indexName, true)
 	}
 }
 
-func (c *connector) processMessage(msg Message) bool {
-	if len(c.cfg.Elasticsearch.TableIndexMapping) == 0 {
-		return true
+func (c *connector) resolveTableToIndexName(fullTableName, tableNamespace, tableName string) string {
+	tableIndexMapping := c.cfg.Elasticsearch.TableIndexMapping
+	if len(tableIndexMapping) == 0 {
+		return fullTableName
```

If there’s no table-to-index mapping, I think we shouldn't return a full table name. Since the expected behavior is that the app should panic when an index name isn’t explicitly specified (index mapping is empty and action has not any index name), silently continuing might lead to unexpected results. What do you think?

---

Yes, you are right, I missed this while refactoring.